flink-core/src/main/java/org/apache/flink/api/common/operators/MailOptionsImpl.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.api.common.operators;
+
+import org.apache.flink.annotation.Internal;
+
+/** Options to configure behaviour of executing mailbox mails. */
+@Internal
+public class MailOptionsImpl implements MailboxExecutor.MailOptions {
+    private boolean interruptible;
+
+    @Override
+    public MailboxExecutor.MailOptions setInterruptible() {
+        this.interruptible = true;
+        return this;
+    }
+
+    public boolean isInterruptible() {
```

Comment:
Why is this method defined in class and not in (super) interface?

Reply:
I've done it to both:
- minimise the size of the public API from our maintenance perspective
- do not confuse user which methods to use/which direction (that's a builder, not pojo to read values) `MailOptions` should work. User should never call `isInterruptible()`.